### PR TITLE
bugfix - keep directory to deactivated environment in PATH

### LINKED ISSUE #16343: If an environment's view's bin folder is in the user's PATH variable while the environment is not active, `spack env activate <name> && spack env deactivate` will remove it.
If an environment's view's bin folder is in the user's PATH variable while the environment is not active, `spack env activate <name> && spack env deactivate` will remove it.

My use case: I have a set of development tools I have installed via spack in an environment with a view in my home directory. If I need to add a new tool, I'll activate the environment, install the tool, and then deactivate it. Unfortunately, this also removes the directory to the view's bin dir from my PATH.

I don't want to activate the environment in my profile because 1) it can be slow, and 2) I may be using another environment.

### Steps to reproduce the issue

```console
$ spack env create --with-view $HOME/foo-view foo
$ export PATH=$HOME/foo-view/bin:$PATH
$ spack env activate foo
$ spack env deactivate
$ echo $PATH
# Observe that $HOME/foo-view/bin is no longer in $PATH
...
```

### Error Message
N/A

### Information on your system

* **Spack:** 0.14.2
* **Python:** 3.7.7
* **Platform:** darwin-catalina-skylake

### Additional information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have searched the issues of this repo and believe this is not a duplicate
- [x] I have run the failing commands in debug mode and reported the output (No failing commands)

### PR DESCRIPTION
**Pull request by RikkiButler20**
Resolves [#16343](https://github.com/spack/spack/issues/16343)

Keeping directory to deactivated environment in PATH if the user manually added the directory.

### CODE REVIEW

File: lib/spack/spack/test/cmd/env.py [hunk 1]
```diff
@@ -1030,6 +1030,35 @@ def noop(*args):
         assert x_read['z'] != y_read['z']
 
 
+def test_env_dir_stays_after_deactivation():  # TODO: do docs
```
Comment (becker33) [inline lib/spack/spack/test/cmd/env.py]: If you use the `working_env` fixture as an argument to the test, then you don't have to worry about your test polluting the user environment for other tests.

```suggestion
def test_env_dir_stays_after_deactivation(working_env):  # TODO: do docs
```

**Comment by becker33** [comment]
@tgamblin can you review this since @RikkiButler20 and I both wrote code for it?

**Comment by alecbcs** [comment]
@becker33 @RikkiButler20 are you all still waiting on a review from @tgamblin here? Or is this bugfix no longer relevant?

**Comment by alecbcs** [comment]
Closing with no response. @RikkiButler20 @tgamblin @becker33 feel free to reopen if this is still something you all want to work on.

**Pull request closed without merging**